app/controllers/applications_controller.rb

```diff
@@ -4,25 +4,24 @@ def new
     @application = Application.new
   end
 
-  def create 
+  def create
     application = Application.create(app_params)
-    pet_ids = params[:pet_names].map {|id| id.to_i}
-    pets = pet_ids.map {|id| Pet.find(id)}
-    pets.each {|pet| pet.applications << application}
-    pets.each {|pet| ApplicationPet.create(pet_id: pet.id, application_id: application.id)}
```

Comment:
We had to remove these as we were duplicating applications and pets each time. When creating an application, we only have to either: make a new ApplicationPet object, add pets to an application object, or add application to a pet object.